tardis/plasma/properties/nlte_rate_equation_solver.py

```diff
@@ -699,3 +711,202 @@ def prepare_solution_vector(self, number_density):
             )
         solution_vector = np.hstack(solution_array + [0])
         return solution_vector
+
+    @staticmethod
+    def prepare_coefficient_matrices_excitation(
```

Comment:
I don't quite understand what this method does. It would be useful to expand upon the description in the docstring (including e.g. shape information for the created DataFrame).